src/source/Ice/IceAgent.c

```diff
@@ -364,12 +364,24 @@ STATUS iceAgentGatherLocalCandidate(PIceAgent pIceAgent)
     KvsIpAddress localIpAddresses[MAX_LOCAL_NETWORK_INTERFACE_COUNT];
     UINT32 localIpAddressesCount = MAX_LOCAL_NETWORK_INTERFACE_COUNT, i;
     PSocketConnection pSocketConnection = NULL;
+    BOOL filterCallbackSet = TRUE;
 
-    CHK_STATUS(getLocalhostIpAddresses(localIpAddresses, &localIpAddressesCount));
+    // Check if a callback is set for filtering network interfaces
+    if(pIceAgent->kvsRtcConfiguration.setFilterFn == NULL) {
```

Comment:
Is it possible to move this all to https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/src/source/Ice/Network.c#L27 ?

Then we can just do `continue` and not have to copy stuff around.

Reply:
Done